lib/pages/movies_page.dart

```diff
@@ -107,7 +106,23 @@ class _MoviesPageState extends State<MoviesPage> {
     );
   }
 
-  Card _buildGenreCard(String genre) {
+  List<Widget> _buildGenresList({required List<int> genres}) {
+    final children = <Widget>[];
+    final transformedGenres = <Genre>[];
+    transformedGenres.addAll(
+      GenreModel.validGenres.where(
+        (g) => genres.contains(
+          g.id,
+        ),
+      ),
+    );
+    for (Genre genre in transformedGenres) {
+      children.add(_buildGenreCard(genre));
+    }
+    return children;
+  }
```

Comment:
i think we can find a more efficient way to build this method using a map

Reply:
done!